src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -99,29 +103,52 @@ const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
   }
 }
 
-export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
-  <TouchableHighlight onPress={onSelected} underlayColor={colors["gray-light"]}>
-    <Container>
-      <Icon source={require("../../../../../../images/payment_request.png")} />
-      <TextContainer>
-        <PaymentRequest>Payment request</PaymentRequest>
-        <CostLabel>
-          {invoice.total}
-        </CostLabel>
-      </TextContainer>
-      <TextContainer>
-        <InvoiceStateButton state={invoice.state} />
-      </TextContainer>
-    </Container>
-  </TouchableHighlight>
-
-export default createFragmentContainer(
+export const InvoicePreview: React.SFC<Props> = ({ conversationId, relay, invoice, onSelected }) => {
+  const { ARNotificationsManager } = NativeModules
+  const notificationsManagerEmitter = new NativeEventEmitter(ARNotificationsManager)
+
+  const paymentRequestPaidCallback = notification => {
+    if (notification.url === invoice.payment_url) {
+      relay.refetch({ conversationId, invoiceId: invoice.lewitt_invoice_id }, null, null, { force: true })
+    }
+  }
+  const subscription = notificationsManagerEmitter.addListener("PaymentRequestPaid", paymentRequestPaidCallback)
```

Comment:
> Don't forget to unsubscribe, typically in `componentWillUnmount`

The documentation has this snippet, so I think you’ll have to make this a class and ensure a component does not receive events anymore once it is unmounted.

Reply:
Updated and verified 💚